test/helpers/with-server.ts

```diff
@@ -4,25 +4,24 @@ import is from '@sindresorhus/is';
 import http = require('http');
 import tempy = require('tempy');
 import createHttpsTestServer, {ExtendedHttpsTestServer, HttpsServerOptions} from './create-https-test-server';
-import createTestServer = require('create-test-server');
+import createHttpTestServer, {ExtendedHttpTestServer, HttpServerOptions} from './create-http-test-server';
 import FakeTimers = require('@sinonjs/fake-timers');
-import got, {InstanceDefaults} from '../../source';
-import {ExtendedGot, ExtendedHttpServer, ExtendedTestServer, GlobalClock, InstalledClock} from './types';
+import got, {InstanceDefaults, Got} from '../../source';
+import {ExtendedHttpServer, GlobalClock, InstalledClock} from './types';
 
-export type RunTestWithServer = (t: test.ExecutionContext, server: ExtendedTestServer, got: ExtendedGot, clock: GlobalClock) => Promise<void> | void;
-export type RunTestWithHttpsServer = (t: test.ExecutionContext, server: ExtendedHttpsTestServer, got: ExtendedGot) => Promise<void> | void;
+export type RunTestWithServer = (t: test.ExecutionContext, server: ExtendedHttpTestServer, got: Got, clock: GlobalClock) => Promise<void> | void;
+export type RunTestWithHttpsServer = (t: test.ExecutionContext, server: ExtendedHttpsTestServer, got: Got, fakeTimer?: GlobalClock) => Promise<void> | void;
 export type RunTestWithSocket = (t: test.ExecutionContext, server: ExtendedHttpServer) => Promise<void> | void;
 
-const generateHook = ({install, options: testServerOptions}: {install?: boolean; options?: unknown}): test.Macro<[RunTestWithServer]> => async (t, run) => {
+const generateHook = ({install, options: testServerOptions}: {install?: boolean; options?: HttpServerOptions}): test.Macro<[RunTestWithServer]> => async (t, run) => {
 	const clock = install ? FakeTimers.install() : FakeTimers.createClock() as GlobalClock;
 
 	// Re-enable body parsing to investigate https://github.com/sindresorhus/got/issues/1186
-	const server = await createTestServer(is.plainObject(testServerOptions) ? testServerOptions : {
+	const server = await createHttpTestServer(is.plainObject(testServerOptions) ? testServerOptions : {
 		bodyParser: {
 			type: () => false
-		},
-		certificate: 'example.com'
-	}) as ExtendedTestServer;
+		} as any
```

Comment:
Why `any`?

Reply:
It was an easy fix, when I was working on it I had a problem with the typing of `bodyParser` and had no time to fix it.

I left it as `any` because that part is going to be changed in the "General improvements of the Test Environment" PR that I'm going to open after the merging of this.